src/axom/slic/docs/sphinx/sections/architecture.rst

```diff
@@ -0,0 +1,440 @@
+
+   * A Logger consists of *four* log message levels: ``ERROR``,
+     ``WARNING``, ``INFO`` and ``DEBUG``.
+
+   * Each :ref:`logMessageLevel` can have one or more :ref:`logStream` instances,
+     which, specify the output destination, format and filtering of
+     messages.
+
+#. One or more :ref:`logStream` instances, bound to a particular logger, that
+   can be shared between log message levels.
+
+The application logs messages at an appropriate :ref:`logMessageLevel`, i.e.,
+``ERROR``, ``WARNING``, ``INFO``, or, ``DEBUG`` using the static API.
+Internally, the Logger routes the message to the corresponding
+:ref:`logStream` instances that are bound to the associated
+:ref:`logMessageLevel`.
+
+The following sections discuss some of these concepts in more detail.
+
+.. _logMessageLevel:
+
+Log Message Level
+^^^^^^^^^^^^^^^^^^
+
+The :ref:`logMessageLevel` indicates the severity of a message. Slic provides
+four levels of messages ranked from highest to lowest as follows:
+
++---------------------+--------------------------------------------------------+
+|   Message Level     |            Usage Description                           |
++=====================+========================================================+
+| ``ERROR``           |  Indicates that the application encountered a critical |
+|                     |  error or a faulty state. Includes also stacktrace.    |
++---------------------+--------------------------------------------------------+
+| ``WARNING``         |  Indicates that the application encountered an error,  |
+|                     |  but, the application should proceed.                  |
++---------------------+--------------------------------------------------------+
+| ``INFO``            |  General information reported by an application.       |
++---------------------+--------------------------------------------------------+
+| ``DEBUG``           |  Information useful to application developers.         |
++---------------------+--------------------------------------------------------+
+
+.. note::
+
+  ``ERROR`` messages by default will cause the application to abort. This
+  behavior may be toggled by calling ``slic::enableAbortOnError()`` and
+  ``slic::disableAbortOnError()``. See the `Slic Doxygen API Documentation`_
+  for more details.
+
+An application may adjust at runtime the severity level of messages to capture
+by calling ``slic::setLoggingMsgLevel()``. For example, the following code
+snippet, sets the severity level to ``WARNING``
+
+.. code-block:: c++
+
+    slic::setLoggingMsgLevel( slic::message::Warning );
+
+This indicates that all messages with a level of severity of ``WARNING`` and
+higher will be captured, namely ``WARNING`` and ``ERROR`` messages. Thereby,
+enable the application to filter out messages with lower severity.
+
+.. _logStream:
+
+Log Stream
+^^^^^^^^^^^
+
+The :ref:`logStream` class, is an abstract base class that facilitates the
+following:
```

Comment:
The tenses here do not agree -- e.g. "Logstream facilitates ... ~~specifies~~ specifying ..."